# R2D2BA #
------------
R2D2BA pipeline allows you to follow allele variations between reference  and RTG genomes.

This script extracts information from VCF files to give a human readable 
files.  It summarizes allele variations, and defines regions of recombination.

##Requirements & Obtaining ##
------------

* Python 2.7 is recommended. 
* Varient Calling Files (VCF) ([More informations about VCF file format](http://www.1000genomes.org/wiki/Analysis/vcf4.0)).
* To download R2D2BA, please use git to download the most recent development
tree.  
* Currently, the tree is hosted on Bitbucket, and can be obtained via:

```
#!bash
git clone https://Aminacurie@bitbucket.org/TardebAnima/meiogenix.git

```
## How do I get started? ##
------------

General help about the execution can be acquired via:

```
#!python

python RUNDIR.py -h
```
It requires 2 inputs:

* VCF file of the reference strain. 
* VCF file of the RTG (this could be replaced by a repository containing more then one RTG).

### Launching ###

Follow the instruction bellow:

```
#!bash

cd path/to/R2D2BA/repository

python RUNDIR.py -i path_to_RTG_VCFfile -j path_to_reference_VCFfile -o path_to_result_repository

```
 
> Results repository should be empty otherwise it is crushed at each run.

> To use more then one RTG_VCFfile you should gather them in one repository and choose the option -I 

### Output file description ###

* -ParentalAlleles.txt : File summarize all the alleles present in the VCF file of the reference strain (reference_VCFfile). 
* -RTGAlleles.txt : File summarize all the alleles presents in the VCF file of the RTG strain (RTG_VCFfile).
* -InvarAllelels.txt : File with alleles that the ratio does not vary between RTGs and reference strain.  
* -VarAlleles.txt : File with alleles that the ratio had varied between RTGs and reference strain.
* -RR.txt : Recombination region file, it extracts all the possible recombined region.



# Depth Of Coverage (DOC)#
------------
The depth of coverage analyzers of the Genome Analysis Toolkit ([GATK](https://software.broadinstitute.org/gatk/gatkdocs/org_broadinstitute_gatk_tools_walkers_coverage_DepthOfCoverage.php)) measure sequence coverage by processing a set of bam files. 
This script normalize the depth of covrage regarding the Run total depth of coverage and regarding the reference. The results are stocked in the file RTG__normalizedXKb.txt.
Results representation is allowed by the Alleles_depth_of_Covrage.R script

##Requirements & Obtaining ##
------------

* Python 2.7 is recommended. 
* R 3.2.3 is recommended.
* depth of coverage files generated by DepthOfCoverage analyzers.

## How do I get started? ##
------------

General help about the execution can be acquired via:

```
#!python
 cd PATH/TO/Depth_Of_Coverage
 python Depth-Of-Cov.py -h
```

### Launching ###

Follow the instruction bellow:

```
#!python

 cd PATH/TO/Depth_Of_Coverage
 python Depth-Of-Cov.py -i path_to_reference_DOCfile -j path_to_reference_RTG_DOCfile -o path_to_result_repository

```
 
> Results repository should be empty otherwise it is crushed at each run.

## Graphical representation ##
------------

```
#!python

 cd PATH/TO/Depth_Of_Coverage
 cat  Alleles_depth_of_Covrage.R | /usr/local/bin/R --slave --args PATH_TO_RTG__normalizedXKb.txt. PATH_TO_figout.pdf

``` 






## Contribution guidelines ##
------------

* Writing tests
* Code review
* Other guidelines


## Who do I talk to? ##
------------

The best way to get help regarding R2D2BA tool is to email the mailing list:

xx@curie.com